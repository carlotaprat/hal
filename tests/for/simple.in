for i in v:
    this
    that


for i in li:
    for j in lj:
        whatever