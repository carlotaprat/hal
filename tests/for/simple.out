digraph {

	ordering=out;
	ranksep=.4;
	bgcolor="lightgrey"; node [shape=box, fixedsize=false, fontsize=12, fontname="Helvetica-bold", fontcolor="blue"
		width=.25, height=.25, color="black", fillcolor="white", style="filled, solid, bold"];
	edge [arrowsize=.5, color="black", style="bold"]

  n0 [label="BLOCK"];
  n1 [label="FOR_STMT"];
  n1 [label="FOR_STMT"];
  n2 [label="PARAMS"];
  n2 [label="PARAMS"];
  n3 [label="i"];
  n4 [label="FUNCALL"];
  n4 [label="FUNCALL"];
  n5 [label="v"];
  n6 [label="ARGS"];
  n7 [label="BLOCK"];
  n7 [label="BLOCK"];
  n8 [label="FUNCALL"];
  n8 [label="FUNCALL"];
  n9 [label="do"];
  n10 [label="ARGS"];
  n10 [label="ARGS"];
  n11 [label="FUNCALL"];
  n11 [label="FUNCALL"];
  n12 [label="this"];
  n13 [label="ARGS"];
  n14 [label="FUNCALL"];
  n14 [label="FUNCALL"];
  n15 [label="do"];
  n16 [label="ARGS"];
  n16 [label="ARGS"];
  n17 [label="FUNCALL"];
  n17 [label="FUNCALL"];
  n18 [label="that"];
  n19 [label="ARGS"];
  n20 [label="FOR_STMT"];
  n20 [label="FOR_STMT"];
  n21 [label="PARAMS"];
  n21 [label="PARAMS"];
  n22 [label="i"];
  n23 [label="FUNCALL"];
  n23 [label="FUNCALL"];
  n24 [label="li"];
  n25 [label="ARGS"];
  n26 [label="BLOCK"];
  n26 [label="BLOCK"];
  n27 [label="FOR_STMT"];
  n27 [label="FOR_STMT"];
  n28 [label="PARAMS"];
  n28 [label="PARAMS"];
  n29 [label="j"];
  n30 [label="FUNCALL"];
  n30 [label="FUNCALL"];
  n31 [label="lj"];
  n32 [label="ARGS"];
  n33 [label="BLOCK"];
  n33 [label="BLOCK"];
  n34 [label="dowhatever"];

  n0 -> n1 // "BLOCK" -> "FOR_STMT"
  n1 -> n2 // "FOR_STMT" -> "PARAMS"
  n2 -> n3 // "PARAMS" -> "i"
  n1 -> n4 // "FOR_STMT" -> "FUNCALL"
  n4 -> n5 // "FUNCALL" -> "v"
  n4 -> n6 // "FUNCALL" -> "ARGS"
  n1 -> n7 // "FOR_STMT" -> "BLOCK"
  n7 -> n8 // "BLOCK" -> "FUNCALL"
  n8 -> n9 // "FUNCALL" -> "do"
  n8 -> n10 // "FUNCALL" -> "ARGS"
  n10 -> n11 // "ARGS" -> "FUNCALL"
  n11 -> n12 // "FUNCALL" -> "this"
  n11 -> n13 // "FUNCALL" -> "ARGS"
  n7 -> n14 // "BLOCK" -> "FUNCALL"
  n14 -> n15 // "FUNCALL" -> "do"
  n14 -> n16 // "FUNCALL" -> "ARGS"
  n16 -> n17 // "ARGS" -> "FUNCALL"
  n17 -> n18 // "FUNCALL" -> "that"
  n17 -> n19 // "FUNCALL" -> "ARGS"
  n0 -> n20 // "BLOCK" -> "FOR_STMT"
  n20 -> n21 // "FOR_STMT" -> "PARAMS"
  n21 -> n22 // "PARAMS" -> "i"
  n20 -> n23 // "FOR_STMT" -> "FUNCALL"
  n23 -> n24 // "FUNCALL" -> "li"
  n23 -> n25 // "FUNCALL" -> "ARGS"
  n20 -> n26 // "FOR_STMT" -> "BLOCK"
  n26 -> n27 // "BLOCK" -> "FOR_STMT"
  n27 -> n28 // "FOR_STMT" -> "PARAMS"
  n28 -> n29 // "PARAMS" -> "j"
  n27 -> n30 // "FOR_STMT" -> "FUNCALL"
  n30 -> n31 // "FUNCALL" -> "lj"
  n30 -> n32 // "FUNCALL" -> "ARGS"
  n27 -> n33 // "FOR_STMT" -> "BLOCK"
  n33 -> n34 // "BLOCK" -> "dowhatever"

}

